### ISSUE TITLE
UI Redesign

### ISSUE THREAD

**Issue by synap5e** [overtrack-gg/overtrack-twitch-extension#7]
The current design works well as a proof of concept, but a redesign is needed. 
This issue will serve as the discussion hub for the direction to take the design of this extension.

All feedback is welcome, and a picture is worth a thousand words (even if it's a napkin or mspaint sketch).

Please keep in mind the limitations of twitch extensions.
Currently the extension is designed as a [video-component](https://dev.twitch.tv/docs/extensions/#types-of-extensions) extension. This means it cannot take up the full width of a stream (maximum width is 50%). This maximizes compatibility (only a single full width extension is allowed, as opposed to 2 video-component extensions) and makes positioning of the extension easier for streamers.

**Comment by synap5e** [comment]
For reference, this is the current design 
![twitch_ui](https://user-images.githubusercontent.com/2515062/59920816-df529700-947f-11e9-9437-ca63f107007e.png)
![image](https://user-images.githubusercontent.com/2515062/60371323-06473500-9a4d-11e9-813e-e6ce1b58b4f7.png)

[On a stream](https://www.twitch.tv/videos/402679261)

**Comment by LeSqueed** [comment]
Not in a order of priority.

- When there's double digits there seems to be an overlap with the icon next to it.
- Names would probably look better underneath the picture, it would be more in line with the design choices within OW itself.
- Not sure if image quality, but the text doesn't seem to have smooth lines, thinking that has to do with it being _cursive_ right now.

**Comment by teamcoltra** [comment]
![image](https://user-images.githubusercontent.com/161164/60319787-73af7480-992c-11e9-8979-9843091714f3.png)

I like the Overwatch League layout, we could mimic this a bit with red and blue and instead of % ult next to the name you would have kills / deaths

**Comment by synap5e** [comment]
@teamcoltra 
Replicate it in full width, or just stylistically? 
Full width would require switching the extension to a fullscreen overlay, which encounters the drawbacks in the OP. We also need to leave room on the right for the killfeed, as the built in spectator mode shifts the killfeed down.
I think the choice should be between replicating the tab screen or spectator mode, but we need to get some feedback on which would be preferred by streamers.

**Comment by teamcoltra** [comment]
Just in diagonal box style. Maybe looking at the option of having full width, but for now just diagonal boxes should look nice.

**Comment by synap5e** [comment]
So stacked? Some of the feedback I had for the current UI was that the friendly row should go at the bottom, since that is how tab is displayed. I'm not as sure if that would still apply for specbar style though.

**Comment by teamcoltra** [comment]
![image](https://user-images.githubusercontent.com/161164/60331535-b895d400-9949-11e9-8937-d76308f22315.png)

**Comment by teamcoltra** [comment]
I actually have a pretty comprehensive pull request to do, it does break the css generator though because I don't know python... but I think it's an easy fix :)

**Comment by jess-sio** [comment]
I think the hero icon needs to be more prominent, and the counters probably shouldn't be center aligned. Also keep in mind Mercy has a resurrect counter that wouldn't fit in this design.

**Comment by teamcoltra** [comment]
In addition I have made a PSD file (will be included in my pull request) which can generate all the heroes in red, blue, and dead configurations. 

Which I forgot that I updated the dead configuration to just add .dead to the hero and then you get this graphic
![image](https://user-images.githubusercontent.com/161164/60332041-cac44200-994a-11e9-8c1c-4961ce49173a.png)

**Comment by teamcoltra** [comment]
> I think the hero icon needs to be more prominent, and the counters probably shouldn't be center aligned. Also keep in mind Mercy has a resurrect counter that wouldn't fit in this design.

The counters are not centre aligned they are just left aligned to the icon. I'm working on Mercy's res counter before push. :)

**Comment by teamcoltra** [comment]
Updated and still works with the batch script / PSD

![image](https://user-images.githubusercontent.com/161164/60333695-8aff5980-994e-11e9-9ff6-a7fc1ad76ae4.png)

I can adjust the icon if we think a mercy would have reasonably more than 9 res in a match, if it goes above it it doesn't really break just peaks over a bit.

![image](https://user-images.githubusercontent.com/161164/60333781-c568f680-994e-11e9-9b14-d8e84d5e0c6e.png)

As a former mercy main myself having 9+ res in a game is pretty crazy considering the long cool down.

**Comment by teamcoltra** [comment]
With no wings on the up arrow I can move it over a bit.

**Comment by synap5e** [comment]
Here's my take on the redesign. Using @teamcoltra's idea of replicating the OWL/spectator bar.
I went with a more CSS heavy approach instead of compositing the background shape, icons, and hero image in one image with photoshop.
I also put the names on the bottom and heroes on the right to replicate how OW does it (also the skew is identical to OW). I kept with using Big Noodle as the font, but I am not sure if this is best as it's not the fastest to read, and deviates from OW's design.

![image](https://user-images.githubusercontent.com/2515062/60411008-f3845a00-9c1e-11e9-890a-dd5ba0183410.png)

I'm also not 100% happy with how mercy resurrects show, going over the top of the image, but I feel it's good enough for now and we can look into improvements later.
![image](https://user-images.githubusercontent.com/2515062/60411130-7a393700-9c1f-11e9-8368-2907e16df27d.png)

Personally I think my version should be merged (#9), then further improvements can be built on this version.
